config/kdump.py

```diff
@@ -90,3 +95,110 @@ def kdump_num_dumps(db, kdump_num_dumps):
     check_kdump_table_existence(kdump_table)
 
     db.cfgdb.mod_entry("KDUMP", "config", {"num_dumps": kdump_num_dumps})
+    echo_reboot_warning()
+
+
+# 'remote' command ('sudo config kdump remote ...')
+@kdump.command(name="remote", short_help="Enable or Disable Kdump Remote")
+@click.argument('action', required=True, type=click.Choice(['enable', 'disable'], case_sensitive=False))
+@pass_db
+def kdump_remote(db, action):
+
+    """Enable or Disable Kdump Remote Mode"""
+    kdump_table = db.cfgdb.get_table("KDUMP")
+    check_kdump_table_existence(kdump_table)
+    current_remote_status = kdump_table.get("config", {}).get("remote", "false").lower()
+
+    if action.lower() == 'enable' and current_remote_status == 'true':
+        click.echo("Error: Kdump Remote Mode is already enabled.")
+        return
+
+    elif action.lower() == 'disable' and current_remote_status == 'false':
+        click.echo("Error: Kdump Remote Mode is already disabled.")
+        return
+
+    if action.lower() == 'disable':
+        ssh_string = kdump_table.get("config", {}).get("ssh_string", None)
+        ssh_key = kdump_table.get("config", {}).get("ssh_key", None)
+        if ssh_string or ssh_key:
+            click.echo("Error: Remove SSH_string, SSH_key from Config DB before disabling Kdump Remote Mode.")
+            return
+
+    remote = 'true' if action.lower() == 'enable' else 'false'
+    db.cfgdb.mod_entry("KDUMP", "config", {"remote": remote})
+    file_path = Path('/etc/default/kdump-tools')
+
+    # Values to be set for SSH and SSH_KEY
+    DEFAULT_SSH = "<user at server>"
+    DEFAULT_SSH_KEY = "<path>"
+
+    with open(file_path, 'r') as file:
+        lines = file.readlines()
+    updated_lines = []
+
+    for line in lines:
+        if remote:
+            if line.startswith("#SSH="):
+                updated_lines.append(f'SSH={DEFAULT_SSH}\n')
+            elif line.startswith("#SSH_KEY="):
+                updated_lines.append(f'SSH_KEY={DEFAULT_SSH_KEY}\n')
+            else:
+                updated_lines.append(line)
+        else:
+            if line.startswith("SSH="):
+                updated_lines.append(f'#SSH={DEFAULT_SSH}\n')
+            elif line.startswith("SSH_KEY="):
+                updated_lines.append(f'#SSH_KEY={DEFAULT_SSH_KEY}\n')
+            else:
+                updated_lines.append(line)
+
+    with open(file_path, 'w') as file:
+        file.writelines(updated_lines)
```

Comment:
Can't this be handled in hostcfgd?

Reply:
You're correct; this could be managed by hostcfgd for better consistency. We chose to handle it directly here for simplicity and immediate updates. we will modify the code hostcfgd to handle this and remove from utility